From 6b836c0166fc819040d38447ff1c99470fff5a13 Mon Sep 17 00:00:00 2001
From: Finley Xiao <finley.xiao@rock-chips.com>
Date: Fri, 27 Mar 2020 15:18:23 +0800
Subject: [PATCH] ARM: dts: rockchip: rv1126: Modify npu opp table

Change 400MHz to 396MHz, 396MHz comes from GPLL, and sometimes HPLL can
be closed

vgg16_maxpool pass.
NPU leakage range: 0.6mA to 1.3mA
0.6mA 0.8mA 0.9mA 1mA 1.1mA 1.2mA 1.3mA

Change-Id: I88edb3e687131b42ed6890153711534bd9f6cc0f
Signed-off-by: Finley Xiao <finley.xiao@rock-chips.com>
---
 arch/arm/boot/dts/rv1126.dtsi | 30 +++++++++++++++++++++++++-----
 1 file changed, 25 insertions(+), 5 deletions(-)

diff --git a/arch/arm/boot/dts/rv1126.dtsi b/arch/arm/boot/dts/rv1126.dtsi
index f232f0732d4d..c042fe8e0a70 100644
--- a/arch/arm/boot/dts/rv1126.dtsi
+++ b/arch/arm/boot/dts/rv1126.dtsi
@@ -1698,7 +1698,7 @@
 		clocks = <&cru ACLK_NPU>, <&cru HCLK_NPU>, <&cru PCLK_PDNPU>, <&cru CLK_CORE_NPU>;
 		clock-names = "aclk_npu", "hclk_npu", "pclk_pdnpu", "sclk_npu";
 		assigned-clocks = <&cru CLK_CORE_NPU>;
-		assigned-clock-rates = <400000000>;
+		assigned-clock-rates = <396000000>;
 		operating-points-v2 = <&npu_opp_table>;
 		interrupts = <GIC_SPI 107 IRQ_TYPE_LEVEL_HIGH>;
 		power-domains = <&power RV1126_PD_NPU>;
@@ -1708,13 +1708,33 @@
 	npu_opp_table: npu-opp-table {
 		compatible = "operating-points-v2";
 
-		opp-400000000 {
-			opp-hz = /bits/ 64 <400000000>;
-			opp-microvolt = <800000 800000 1000000>;
+		opp-200000000 {
+			opp-hz = /bits/ 64 <200000000>;
+			opp-microvolt = <725000 725000 1000000>;
+		};
+		opp-300000000 {
+			opp-hz = /bits/ 64 <300000000>;
+			opp-microvolt = <725000 725000 1000000>;
+		};
+		opp-396000000 {
+			opp-hz = /bits/ 64 <396000000>;
+			opp-microvolt = <725000 725000 1000000>;
+		};
+		opp-500000000 {
+			opp-hz = /bits/ 64 <500000000>;
+			opp-microvolt = <725000 725000 1000000>;
 		};
 		opp-600000000 {
 			opp-hz = /bits/ 64 <600000000>;
-			opp-microvolt = <800000 800000 1000000>;
+			opp-microvolt = <725000 725000 1000000>;
+		};
+		opp-700000000 {
+			opp-hz = /bits/ 64 <700000000>;
+			opp-microvolt = <775000 775000 1000000>;
+		};
+		opp-800000000 {
+			opp-hz = /bits/ 64 <800000000>;
+			opp-microvolt = <825000 825000 1000000>;
 		};
 	};
 
-- 
2.35.3

